doc/user_guide/psyke.rst

```diff
@@ -312,9 +339,48 @@ The generated code is now:
         CALL testkern_code_w2_only(nlayers, f3_proxy%data, f2_proxy%data, ndf_w2, undf_w2, map_w2(:,cell))
       END DO
       !$omp end parallel do
+      CALL psy_data%PostStart
+      CALL psy_data%ProvideVariable("cell_post", cell)
+      CALL psy_data%ProvideVariable("df_post", df)
+      CALL psy_data%ProvideVariable("f3_post", f3)
+      CALL psy_data%PostEnd
       !
       ! ExtractEnd
 
+.. note::
+
+    At this stage builtins are not fully supported, resulting in ``f2``
+    being incorrectly detected as an input parameters, and not as an
+    output parameter. This issue is tracked in #637.
+
+
 Examples in ``examples/lfric/eg12`` directory demonstrate how to
 apply code extraction by utilising PSyclone transformation scripts
 (see :ref:`examples` section for more information).
+
+.. _psyke_netcdf:
+
+NetCDF Extraction Example
+-------------------------
+PSyclone comes with an example NetCDF based extraction library in
+``lib/extract/dl_esm_inf/netcdf``
+(https://github.com/stfc/PSyclone/tree/master/lib/extract/dl_esm_inf/netcdf).
+This library implements the full PSyData
+API for use with the GOcean 1.0 dl_esm_inf infrastructure library.
+In order to compile this library, you must have NetCDF installed.
+When running the code, it will create a NetCDF file for the instrumented
+code region. It includes all variables that are read before the code
+is executed, and all variables that have been modified. The output
+variables have the postfix ``_post`` attached to the NetCDF names,
+e.g. a variable ``xyz`` that is read and written will be stored
+with the name ``xyz`` containing the input values, and the name
+``xyz_post`` containing the output values. Arrays have their size
+stored as NetCDF dimensions: again the variable ``xyz`` will have its
+sizes stored as ``xyzdim1``, ``xyzdim2`` for the input values,
+and output arrays use the name ``xyz_postdim1``, ``xyz_postdim2``.
+
+The output file only stores the values used in the subroutine,
+e.g. any GOcean grid properties (see :ref:`gocean1.0-grid-props`)
```

Comment:
Sphinx warns that this label (gocean1.0-grid-props) is undefined.

Reply:
The label was actually defined, but not at the beginning of a section, so sphinx couldn't create a reference. After a bit of reading I added a table number, so that this becomes a reference to the table, which is much better than referencing a whole section.